tests/src/kakarot/accounts/test_contract_account.py

```diff
@@ -20,52 +20,74 @@ def bytecode(self, request):
         return random.randbytes(request.param)
 
     class TestInitialize:
+        @SyscallHandler.patch("IKakarot.register_account", lambda addr, data: [])
         @SyscallHandler.patch("IKakarot.get_native_token", lambda addr, data: [0xDEAD])
         @SyscallHandler.patch("IERC20.approve", lambda addr, data: [1])
-        @SyscallHandler.patch("Account_kakarot_address", 0x1234)
-        @SyscallHandler.patch("Account_evm_address", 0xABDE1)
-        def test_should_store_given_addresses(self, cairo_run):
+        def test_should_store_given_addresses_and_initialize(self, cairo_run):
             cairo_run(
-                "test__initialize__should_store_given_evm_address",
+                "test__initialize",
+                kakarot_address=0x1234,
+                evm_address=0xABDE1,
                 implementation_class=0xC1A55,
             )
+            SyscallHandler.mock_storage.assert_any_call(
+                address=get_storage_var_address("Account_evm_address"), value=0xABDE1
+            )
+
             SyscallHandler.mock_storage.assert_any_call(
                 address=get_storage_var_address("Ownable_owner"), value=0x1234
             )
             SyscallHandler.mock_storage.assert_any_call(
-                address=get_storage_var_address("Account_evm_address"), value=0xABDE1
+                address=get_storage_var_address("Account_is_initialized"), value=1
             )
 
+        @SyscallHandler.patch("IKakarot.register_account", lambda addr, data: [])
+        @SyscallHandler.patch("IKakarot.get_native_token", lambda addr, data: [0xDEAD])
+        @SyscallHandler.patch("IERC20.approve", lambda addr, data: [1])
+        def test_should_store_implementation(self, cairo_run):
```

Comment:
why not one single test `test_should_set_storage_variables`?

Reply:
that's what the existing test design was like. I merged them.